client/document.go

```diff
@@ -886,7 +939,11 @@ func (doc *Document) setDocID(docID DocID) {
 	doc.id = docID
 }
 
-// generateAndSetDocID generates the DocID and then (re)sets `doc.id`.
+// GenerateAndSetDocID generates the DocID and then (re)sets `doc.id`.
+func (doc *Document) GenerateAndSetDocID() error {
```

Comment:
todo: Code-cov now says this is untested, and I couldn't spot any references to it.  The documentation is probably not clear enough for users to know when they should be calling this.

Please either remove this function, or expand the documentation if you think users should be calling this.

Reply:
That a bug with codecov because it is definitely used. It used anytime a new embedding is generated on document creation. This is what ensure the docID is representative of the contents.

Reply:
The environment variable in the workflow had a typo. The embedding related test where not executed which affected code coverage.